### ISSUE TITLE
App icon not displaying in expo go

### ISSUE THREAD

**Issue by g4laura** [expo/expo#20143]
### Summary

I'm using the managed workflow and defining my icons in app.json, but they are not showing up in the expo go app during local testing or eas builds.

### What platform(s) does this occur on?

Android, iOS

### Environment

    System:
      OS: macOS 12.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 16.13.2 - /usr/local/bin/node
      Yarn: 1.22.19 - /usr/local/bin/yarn
      npm: 8.19.2 - /usr/local/bin/npm
      Watchman: 2022.10.31.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
      Android SDK:
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8609683
      Xcode: 14.1/14B47b - /usr/bin/xcodebuild
    npmPackages:
      @expo/webpack-config: ^0.17.0 => 0.17.2 
      expo: ^46.0.0 => 46.0.16 
      react: 18.0.0 => 18.0.0 
      react-dom: 18.0.0 => 18.0.0 
      react-native: 0.69.6 => 0.69.6 
      react-native-web: ~0.18.7 => 0.18.9 
    npmGlobalPackages:
      eas-cli: 2.7.1
      expo-cli: 6.0.6
    Expo Workflow: managed

### Minimal reproducible example

 "expo": {
  
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "userInterfaceStyle": "light",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff"
    },
}

**Comment by brentvatne** [comment]
hi there! can you elaborate on where in expo go you are not seeing this? the app icon can be cached in expo go and so that may be what you are experiencing. it's hard to know without more info. if you can post a reproducible example project that we can clone and more information about where you are expecting to see the app icon that you are not, then we can verify this for you.

**Comment by convurt** [comment]
I cant get the icon to update regardless of what I try.  I think I originally built the first build without an icon it's just blank).  Then i've gone back and added an icon, and it will not update to the new icon. I tried:
eas update:configure
eas build --profile preview --platform ios --clear-cache
but nothing will grab the icon and pull it in.  I have it set 1024x1024.  A screenshot of the icon is attached. Any idea what I could try?


<img width="205" alt="Screen Shot 2023-01-07 at 8 02 00 AM" src="https://user-images.githubusercontent.com/98235957/211152008-97ed2084-4832-4ed2-ba9f-e985385b4d18.png">

    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "cover",
      "backgroundColor": "#000000"
    },

**Comment by dmitryame** [comment]
Same here, this is annoying, especially if you have multiple projects to manage. Are we going to hear anything from the expo support team on this?

**Comment by miamiappmachine** [comment]
Same here on one project.

**Comment by sumitsharma7667** [comment]
Same with me. even after build icon is still missing.

**Comment by Overdash** [comment]
@brentvatne I believe this issue has enough information now? (there's a screenshot with the relevant code).
Can this issue be re-opened? Or at least direct those of us affected to a doc that can explain how to show your project's logo in the Expo Go home screen?

**Comment by brentvatne** [comment]
@Overdash - we require a minimal reproducible example. please share a repository we can clone in a new issue.

also, this is likely an ios caching issue. try restarting your device.

**Comment by Overdash** [comment]
This is happening on iOS and Android - I'll aim to provide a repo if OP has abandoned the issue :)

**Comment by brentvatne** [comment]
thanks @Overdash!

**Comment by Arkwell** [comment]
Try just deleting the app entirely and re-downloading and installing from TestFlight. If the icon updates in TestFlight or pretty much anywhere else, you may have to just refresh the app.

**Comment by rickyanwar** [comment]
same here, my project not update icon after update SDK :"expo": "^49.0.0",
and use EAS build but splash screen change only icon not change

this my app.json
```
{
  "expo": {
    "name": "Mostman",
    "slug": "Mostamn",
    "version": "1.0.0",
    "privacy": "public",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "userInterfaceStyle": "light",
    "splash": {
      "image": "./assets/images/splash/splash.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0,
      "url": "https://u.expo.dev/86dc327a-8fb6-4148-b35e-085657554da3"
    },
    "plugins": [
      [
        "expo-image-picker",
        {
          "photosPermission": "The app accesses your photos to let you share them with your friends."
        }
      ]
    ],
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#FFFFFF"
      },
      "notification": {
        "icon":"./assets/icons/bus.png"
      },
      "package": "com.mostman.app",
      "googleServicesFile": "./google-services.json"
    },
    "web": {
      "favicon": "./assets/favicon.png"
    },
    "extra": {
      "eas": {
        "projectId": "86dc327a-8fb6-4148-b35e-085657554da3"
      }
    },
    "runtimeVersion": {
      "policy": "sdkVersion"
    },
    "useClassicUpdates":true
  }
}

```

file also exists and file size is 1024x1024
![image](https://github.com/expo/expo/assets/36995966/9c7ae593-dc7b-4808-a44f-518d3e18e7f1)

 "icon": "./assets/icon.png",

**Comment by Four58** [comment]
here is my repo: https://github.com/Four58/test-ios-icon-reactnative-expo

I barely touch the project and still I can't see the icon. 
![image](https://github.com/expo/expo/assets/74414586/be864cac-9e6e-494d-be1a-bd4d23d4f952)
![image](https://github.com/expo/expo/assets/74414586/a417e6f2-530d-4b0d-996f-3846b1819b90)
@brentvatne please tell me if you do need more information or if i misunderstand something.

**Comment by John-Mwanzia** [comment]
On your app.json file, change the "foreground-image" from "./assets/adaptive-icon" to your icon's relative file path (just like you did on the "icon" key)

**Comment by sebastianbarcia** [comment]
es lo mismo Jhon no funciona
probe de todo pero no puedo hacer que el icono se muestre

**Comment by Aqibiweb** [comment]
Same here, the icon does not show as I change the icon path with a new one. Developer menu only show text without icon.

**Comment by ldsmoreira** [comment]
Same here

**Comment by ONeill45** [comment]
I am also having this issue, no icon is displayed at all

**Comment by badhrin** [comment]
Same issue here on Android Expo Preview (Distribution) builds. I have tried changing the name of the Adaptive Icon PNG. Also tried --clear-cache option.  Nothing is able to change the app icon at all.

**Comment by rabume** [comment]
For me, the issue was that the images were "cached" in the 'android' folder. After deleting the 'android' folder, the updated icons appeared. I don't know why the icons don't get overwritten when they are changed, though. I suppose this is only a fix if you are using a bare workflow.

**Comment by Chamesss** [comment]
The issue is that Expo doesn't update the android build after making changes in app.json. Delete the ./android build folder and run `npx expo run:android`. That should solve the problem

**Comment by badhrin** [comment]
> The issue is that Expo doesn't update the android build after making changes in app.json. Delete the ./android build folder and run `npx expo run:android`. That should solve the problem.

@Chamesss  Thanks for your response. But I am running the build on eas servers and not locally. So the "run android" doesnt even come in the picture for me. Any thing else i can do ?

**Comment by bigkid-dev** [comment]
@Chamesss or anyone else concerned i experienced a similar issue and what i did to resolve is to move the app icon into another folder and reference the new folder in the app.json file.. hope this helps

**Comment by robimuller** [comment]
im having the same issue at the moment, will get back to you guys if it something works out.

**Comment by robimuller** [comment]
okay, I think I was just really confused, and made an amateur mistake I guess. Here is what my problem was. 

I am relatively new to app dev, and didnt really understand the difference between expo managed workflow and bare workflow. 

I build my app with expo managed workflow in the beginning but ejected later because of firebase, phone auth updates... 

then I kept trying to edit the app.json file with the images, yet the icon didnt update anywhere. Long story short: 

if you eject: delete your app from the simulator. then you have to natively edit the image assets: for instance on iOS get in your project root, -> terminal: open ios/projectname xcworkspace

go to the project file in Xcode, /mages - and add all your icons. 

then run expo run:ios and voila the icon is there. This only took me 5 hrs of headache.

**Comment by Anirudh596** [comment]
hey folks, I got the easy solution create a new images folder in the /assets folder and add the all icons and used assets in it.
after this update the app.json where the images are used so it can render the images.
after that please make sure to rebuild or reload the app then it shows the icons and splash screen.

Happy Coding folks 😊, until AI takes over 💀

**Comment by ruru-m07** [comment]
@Anirudh596 like this?

![image](https://github.com/expo/expo/assets/142723369/afba7568-c02d-4cd2-9c3c-45f070023419)

this still not working...

**Comment by Anirudh596** [comment]
Please rebuild the app and if you already did then please share the app.json file code so i can see through the issue.

**Comment by ruru-m07** [comment]
@Anirudh596 i fix that issue i just clear catch data on expo app and delete .expo folder that my problem is solved thanks. ❤️

**Comment by ralphjenrey** [comment]
You can also delete the uninstall the expo on your device if your using physical then reinstall. It works for me

**Comment by EduardaGiudice** [comment]
Guys, I managed to solve it as follows, first run the command 'npx expo prebuild --clear'.
Then 'npx expo prebuild'
If you are going to generate an apk with 'eas build', delete the 'android' folder before

**Comment by dmitryame** [comment]
if it’s so easy, why can’t expo just fix it?Sent from my iPhoneOn Apr 18, 2024, at 16:41, Eduarda Giudice ***@***.***> wrote:﻿
Guys, I managed to solve it as follows, first run the command 'npx expo prebuild --clear'.
Then 'npx expo prebuild'
If you are going to generate an apk with 'eas build', delete the 'android' folder before

**Comment by rogeriojacome** [comment]
Same here, I already tried 'npx expo prebuild' and was unsuccessful

**Comment by ikeliec** [comment]
> same here, my project not update icon after update SDK :"expo": "^49.0.0", and use EAS build but splash screen change only icon not change
> 
> this my app.json
> 
> ```
> {
>   "expo": {
>     "name": "Mostman",
>     "slug": "Mostamn",
>     "version": "1.0.0",
>     "privacy": "public",
>     "orientation": "portrait",
>     "icon": "./assets/icon.png",
>     "userInterfaceStyle": "light",
>     "splash": {
>       "image": "./assets/images/splash/splash.png",
>       "resizeMode": "cover",
>       "backgroundColor": "#ffffff"
>     },
>     "updates": {
>       "fallbackToCacheTimeout": 0,
>       "url": "https://u.expo.dev/86dc327assssss"
>     },
>     "plugins": [
>       [
>         "expo-image-picker",
>         {
>           "photosPermission": "The app accesses your photos to let you share them with your friends."
>         }
>       ]
>     ],
>     "assetBundlePatterns": [
>       "**/*"
>     ],
>     "ios": {
>       "supportsTablet": true
>     },
>     "android": {
>       "adaptiveIcon": {
>         "foregroundImage": "./assets/adaptive-icon.png",
>         "backgroundColor": "#FFFFFF"
>       },
>       "notification": {
>         "icon":"./assets/icons/bus.png"
>       },
>       "package": "com.mostman.app",
>       "googleServicesFile": "./google-services.json"
>     },
>     "web": {
>       "favicon": "./assets/favicon.png"
>     },
>     "extra": {
>       "eas": {
>         "projectId": "86dc327a-8aaaaa54da3"
>       }
>     },
>     "runtimeVersion": {
>       "policy": "sdkVersion"
>     },
>     "useClassicUpdates":true
>   }
> }
> ```
> 
> file also exists and file size is 1024x1024 ![image](https://private-user-images.githubusercontent.com/36995966/273406423-9c7ae593-dc7b-4808-a44f-518d3e18e7f1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg5MTc5MjEsIm5iZiI6MTcxODkxNzYyMSwicGF0aCI6Ii8zNjk5NTk2Ni8yNzM0MDY0MjMtOWM3YWU1OTMtZGM3Yi00ODA4LWE0NGYtNTE4ZDNlMThlN2YxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIwVDIxMDcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhNGFhNzc4YzgwNzc0MWNlOTQ3YmIyMDA2MzAzMTQ2ZDIyNDkyMWE1ZWQzYjJhYTUwMDA0YTY3ZmJlYTNlZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.d7FGOtHy5w88413zZoiYQZCK8UR2pgSTISZAGpDrYY4)
> 
> "icon": "./assets/icon.png",

Deleting the android folder and trying again, would have worked.

**Comment by andersonguestart098** [comment]
Pessoal pra mim funcionou usando:

felipetartari@MacBook-Pro-de-Felipe ReactIos % expo prebuild --clean
! Git branch has uncommited file changes
› It's recommended to commit all changes before proceeding in case you want to revert generated changes.

✔ Continue with uncommited changes? … yes

✔ Cleared android, ios code
✔ Created native directories
✔ Updated package.json | no changes
(node:29291) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Warning: This command requires version ^2.1.0 of `sharp-cli`. 
Currently installed version: "5.1.0" 
Required version: "^2.1.0" 
You can install it using `npm install -g sharp-cli@^2.1.0`. 

For prerequisites, see: https://sharp.dimens.io/en/stable/install/#prerequisites
» android: userInterfaceStyle: Install expo-system-ui in your project to enable this feature.
✔ Finished prebuild
⠴ Installing CocoaPods...
✔ Installed CocoaPods
felipetartari@MacBook-Pro-de-Felipe ReactIos % 
felipetartari@MacBook-Pro-de-Felipe ReactIos % npx expo prebuild
(node:30213) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
✔ Created native directories | reusing /android, /ios
✔ Updated package.json | no changes
Warning: This command requires version ^2.1.0 of `sharp-cli`. 
Currently installed version: "5.1.0" 
Required version: "^2.1.0" 
You can install it using `npm install -g sharp-cli@^2.1.0`. 

For prerequisites, see: https://sharp.dimens.io/en/stable/install/#prerequisites
» android: userInterfaceStyle: Install expo-system-ui in your project to enable this feature.
✔ Finished prebuild
felipetartari@MacBook-Pro-de-Felipe ReactIos % npm start            

> cemearapp@1.0.0 start
> expo start

Starting project at /Users/felipetartari/cemearApp/ReactIos
(node:30453) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Starting Metro Bundler
Recrawled this watch 1 time, most recently because:
MustScanSubDirs UserDroppedTo resolve, please review the information on
https://facebook.github.io/watchman/docs/troubleshooting.html#recrawl
To clear this warning, run:
`watchman watch-del '/Users/felipetartari/cemearApp/ReactIos' ; watchman watch-project '/Users/felipetartari/cemearApp/ReactIos'`

▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █▀█ █▄▄▄▀██ ▄▄▄▄▄ █
█ █   █ █▀▀▀█ ▀██▀█ █   █ █
█ █▄▄▄█ █▀ █▀▀█▄▀▀█ █▄▄▄█ █
█▄▄▄▄▄▄▄█▄▀ ▀▄█▄█ █▄▄▄▄▄▄▄█
█   ▄ ▀▄ ▄▄▀▄▀ ▀▄▄ █▄█ █ ██
██▄▄ █▄▄▀▀█▄█▀██ ▄▀ ▀  █▄ █
█▀▄▀ ▄▄▄▀▀▄▄█▄▄▀▄▄█▄█▄▀ ▀██
█ █▀▄▄ ▄███▄ ▄█▀ ▄▄███ ▄  █
█▄█████▄▄ ▄▄▄▀▀█▄ ▄▄▄ ▄ ▄▄█
█ ▄▄▄▄▄ █▄█▄█▀ ▄▀ █▄█ ▀▄  █
█ █   █ █  ▀█▄█▄  ▄   ▄   █
█ █▄▄▄█ █  ▀ ▄█▀▀ ▄▀▀█▀▄█ █
█▄▄▄▄▄▄▄█▄██▄███▄█▄▄███▄▄▄█

› Metro waiting on exp://10.1.1.117:8081
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Using Expo Go
› Press s │ switch to development build

› Press a │ open Android
› Press i │ open iOS simulator
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu
› Press o │ open project code in your editor

› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
iOS Bundled 3787ms node_modules/expo/AppEntry.js (1420 modules)
 LOG  Carregando fontes e autenticação...
 LOG  Configurando listeners de notificações...
 LOG  Permissões de notificação concedidas.
 LOG  Token de autenticação encontrado: Não
 LOG  Token Expo Push obtido: ExponentPushToken[lHaVxhInav0ZEWR6MXuoQz]
 LOG  Atualizando o badge count...
 WARN  Usuário não autenticado. Não é possível atualizar o badge.
› Opening on iOS...
› Opening exp://10.1.1.117:8081 on iPhone 16 Pro
› Press ? │ show all commands
 LOG  App voltou para o estado ativo. Resetando o badge...
 LOG  Resetando o badge ao abrir o aplicativo...
 LOG  Badge no dispositivo resetado para 0.
 WARN  Usuário não autenticado. Pulando a marcação de notificações como lidas no backend.


olha o fluxo, deu certinho aquii! publiquei agr no apple connect pra comecar os testes <3
